api-server/src/venues/venues.service.ts

```diff
@@ -37,22 +45,26 @@ export class VenuesService {
     });
   }
 
-  create(newVenue: CreateVenueRequest): Promise<VenueModel> {
+  async create(newVenue: CreateVenueRequest): Promise<VenueModel> {
     const id = uuidv4();
     const slug = getSlug(newVenue.name);
 
+    newVenue = await this.fillInGPSCoordinatesIfNeededWhenPossible(newVenue);
+
     return this.venueModel.create({ ...newVenue, id, slug });
   }
 
-  update(id: string, updatedValues: UpdateVenueRequest): Promise<VenueModel> {
-    let values;
-    if (updatedValues.name) {
-      const slug = getSlug(updatedValues.name);
+  async update(
+    id: string,
+    updatedValues: UpdateVenueRequest,
+  ): Promise<VenueModel> {
+    updatedValues = await this.fillInGPSCoordinatesIfNeededWhenPossible(
+      updatedValues,
+    );
 
-      values = { ...updatedValues, slug };
-    } else {
-      values = { ...updatedValues };
-    }
+    const values = isNotNullOrUndefined(updatedValues.name)
```

Comment:
also a bit of scouting, using the ternary ? operator lets this be a bit more concise and allows values to be a constant